### PR TITLE
Remove dynamic API version resolution in outputs

### LINKED ISSUE #143: No Public IP Addresses for location ‘westeurope’ and API version ‘2017-09-01’
[As reported on discuss](https://discuss.elastic.co/t/es-on-azure-version/98645/7)

### PR DESCRIPTION
**Pull request by russcam**
- Change `reference()` functions in outputs to use resource names instead of resource identifiers
- Use resource deployment conditions where possible. Kibana and Jumpbox resources cannot use conditions because outputs from the resource are required, which cannot be conditional, due to Azure validation evaluating all branches. This leads to obscure errors such as

    ```
    Deployment template validation failed: 'The template output reference to 
    'Microsoft.Resources/deployments/jumpbox' requires an API version. Please see 
    https://aka.ms/arm-template for usage details.'.
    ```
- Add support for 5.5.2, 5.5.3 and 5.6.0
- Validate password and ssh parameters.json files against mainTemplate.json input parameters.

Closes #143

### CODE REVIEW

**Comment by russcam** [comment]
Links need to be reverted back to `master` before merging

File: build/tasks/sanity-checks.js [hunk 1]
```diff
@@ -28,6 +28,30 @@ function checks(cb) {
...
+      {
+        name: "password",
+        template: require("../../parameters/password.parameters.json")
+      },
+      {
+        name: "ssh",
+        template: require("../../parameters/ssh.parameters.json")
+      }];
+
+    parameters.forEach(p=> {
```
Comment (Mpdreamz) [inline build/tasks/sanity-checks.js]: ❤️

File: src/mainTemplate.json [hunk 2]
```diff
@@ -1141,16 +1123,7 @@
...
-      "internal": "No",
-      "external": "No",
-      "gateway": "Yes"
-    },
-    "kibanaHttpsOpts": [
-      "No",
-      "Yes"
-    ],
-    "kibanaHttps": "[variables('kibanaHttpsOpts')[mul(length(take(parameters('kibanaKeyBlob'), 1)), length(take(parameters('kibanaCertBlob'), 1)))]]",
+    "kibanaHttps": "[if(and(greater(length(parameters('kibanaKeyBlob')), 0), greater(length(parameters('kibanaCertBlob')), 0)), 'Yes', 'No')]",
```
Comment (Mpdreamz) [inline src/mainTemplate.json]: :heart: hash map lookups :gun:

File: src/partials/node-resources.json [hunk 3]
```diff
@@ -109,7 +100,7 @@
       "properties": {
         "mode": "Incremental",
         "templateLink": {
-          "uri": "[variables('clientTemplateUrl')]",
+          "uri": "[concat(parameters('templateBaseUrl'), 'machines/client-nodes-resources.json')]",
```
Comment (Mpdreamz) [inline src/partials/node-resources.json]: ❤️

**Pull request merged**